webapp/app/js/models/resultsets.js

```diff
@@ -59,32 +58,56 @@ treeherder.factory('ThResultSetModel', [
         'exclusion_state'
     ];
 
-    var registerResultSetPollers = function(){
+    var registerResultSetPollers = function() {
+
+        // these params will be passed in each time we poll to remain
+        // within the constraints of the URL params
+        var rsPollingParams = _.pick($location.search(), rsPollingKeys);
 
-        if( doResultSetPolling() ){
-            // Register resultset poller if it's not registered
-            $interval(function(){
+        // Register resultset poller if it's not registered
+        $interval(function() {
+            // resultset data for the current repository
+            var rsData = repositories[$rootScope.repoName];
 
-                if( (repositories[$rootScope.repoName].resultSets.length > 0) &&
-                    (repositories[$rootScope.repoName].loadingStatus.prepending === false) ){
+            // This case is if we already have at least 1 resultset and we're
+            // polling for more.
+            // If we already have one, and the revision param is passed in,
+            // don't poll for more, because "there can be only one."
 
+            if((rsData.resultSets.length > 0) &&
+               (!rsData.loadingStatus.prepending)) {
+                if (doResultSetPolling(rsPollingParams)) {
```

Comment:
if this comes out false in ``doResultSetPolling`` then we should cancel the interval.